src/app/core/services/api/customer-auth.service.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  HttpClient,
+  HttpErrorResponse,
+  HttpHeaders,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, catchError, tap, throwError } from 'rxjs';
+import { CustomerResponse } from '../../interfaces/interfaces';
+import { SnackbarService } from '../mat-snackbar.service';
+import { environment } from '../../../environment/environment';
+
+export type CustomerRegistrationForm = {
+  email: string;
+  password: string;
+  firstName: string;
+  lastName: string;
+  dateOfBirth: string;
+  addresses: [
+    {
+      streetName: string;
+      city: string;
+      postalCode: string;
+      country: string;
+    },
+  ];
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CustomerAuthService {
+  private apiUrl = `${environment.host}/${environment.project_key}`;
+  private accessToken = localStorage.getItem('AppAccessToken');
+
+  constructor(
+    private http: HttpClient,
+    private snackbarService: SnackbarService,
+  ) {}
+
+  createCustomer(body: CustomerRegistrationForm): Observable<CustomerResponse> {
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${this.accessToken}`,
+    });
```

Comment:
It is better to move out in a private method headers generation and reuse everywhere

Reply:
Or you can have such reusable method in one place as utils and call everywhere you need